src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala

```diff
@@ -66,41 +68,73 @@ class AkkaHttpSecurity(config: Config, sessionStorage: SessionStorage)(implicit
   import AkkaHttpSecurity._
   // TODO: At some point this object should contain the SessionStore (when we implement that)
 
-  val securityLogic: AkkaHttpSecurityLogic =
+  private[pac4j] val securityLogic: AkkaHttpSecurityLogic =
     Option(config.getSecurityLogic) match {
       case Some(v) => v.asInstanceOf[AkkaHttpSecurityLogic]
       case None => new DefaultSecurityLogic[Future[RouteResult], AkkaHttpWebContext]
     }
 
-  val actionAdapter: HttpActionAdapter[Future[RouteResult], AkkaHttpWebContext] =
+  private[pac4j] val actionAdapter: HttpActionAdapter[Future[RouteResult], AkkaHttpWebContext] =
     Option(config.getHttpActionAdapter) match {
       case Some(v) => v.asInstanceOf[HttpActionAdapter[Future[RouteResult], AkkaHttpWebContext]]
       case None => AkkaHttpActionAdapter
     }
 
-  /**
-   * Authenticate using the provided pac4j configuration. Delivers an AuthenticationRequest which can be used for further authorization
-   * this does not apply any authorization ofr filtering.
-   */
-  def withAuthentication(
-               clients: String = null /* Default null, meaning all defined clients */ ,
-               multiProfile: Boolean = true,
-               enforceFormEncoding: Boolean = false, //Force form parameters to be passed for authentication or the request fails
-             ): Directive1[AuthenticatedRequest] =
-    new Directive1[AuthenticatedRequest] {
-      override def tapply(innerRoute: Tuple1[AuthenticatedRequest] => Route): Route = { ctx =>
+  private[pac4j] val callbackLogic: CallbackLogic[Future[RouteResult], AkkaHttpWebContext] =
+    Option(config.getCallbackLogic) match {
+      case Some(v) => v.asInstanceOf[CallbackLogic[Future[RouteResult], AkkaHttpWebContext]]
+      case None => new DefaultCallbackLogic[Future[RouteResult], AkkaHttpWebContext]
+    }
+
+  private[pac4j] def withContext[A](enforceFormEncoding: Boolean): Directive1[AkkaHttpWebContext] =
```

Comment:
Please add comments to this private route

Reply:
done